### PR TITLE
Add HPU Profiler documentation

### PR DESCRIPTION
**Pull request by jerome-habana**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Fixes # (issue).

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/intermediate.rst [hunk 1]
```diff
@@ -97,3 +97,113 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+Profiler will dump traces in json format. The traces can be visualized in 2 ways:
+
+Using PyTorch TensorBoard Profiler
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For further instructions see, https://github.com/pytorch/kineto/tree/master/tb_plugin.
+
+Install tensorboard
+"""""""""""""""""""""
+ python -um pip install tensorboard torch-tb-profiler
```
Comment (Borda) [inline docs/source/intermediate.rst:184]: lets format it as code

File: docs/source/intermediate.rst [hunk 2]
```diff
@@ -97,3 +97,113 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
 
 For more details, please refer to `Memory Stats APIs <https://docs.habana.ai/en/latest/PyTorch/PyTorch_User_Guide/Python_Packages.html#memory-stats-apis>`__.
+
+----
+
+Working with HPUProfiler
+-------------------------
+
+HPUProfiler is a lightning implementation of PyTorch profiler for HPU devices. It aids in obtaining profiling summary of PyTorch functions.
+It subclasses PyTorch Lightning's [PyTorch profiler](https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.profilers.PyTorchProfiler.html#pytorch_lightning.profilers.PyTorchProfiler).
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply docs/source/intermediate.rst:107]:
```suggestion
It subclasses PyTorch Lightning's `PyTorch profiler <https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.profilers.PyTorchProfiler.html#pytorch_lightning.profilers.PyTorchProfiler>`_.
```

File: docs/source/intermediate.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source/intermediate.rst:184]:
```suggestion
.. code-block:: bash

    python -um pip install tensorboard torch-tb-profiler
```

File: docs/source/intermediate.rst [hunk 3]
```diff
@@ -97,3 +97,113 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+
+For further instructions see, https://github.com/pytorch/kineto/tree/master/tb_plugin.
+
+Install tensorboard
+"""""""""""""""""""""
+ python -um pip install tensorboard torch-tb-profiler
+
+Start the TensorBoard server (default at port 6006)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+ tensorboard --logdir ./tensorboard --port 6006
```
Comment (awaelchli) [inline docs/source/intermediate.rst:188]:
```suggestion

.. code-block:: bash
    
    tensorboard --logdir ./tensorboard --port 6006
```

File: docs/source/intermediate.rst [hunk 4]
```diff
@@ -97,3 +97,113 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+
+    # Pass profiler instance to LightningModule
+    profiler = HPUProfiler()
+    model = MyModel(profiler)
+    trainer = Trainer(accelerator="hpu", profiler=profiler)
+
+For more details on profiler, refer to [PyTorchProfiler](https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_intermediate.html)
+
+Visualize Profiled Operations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (awaelchli) [inline docs/source/intermediate.rst:173]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/source/intermediate.rst [hunk 5]
```diff
@@ -97,3 +97,113 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+
+Install tensorboard
+"""""""""""""""""""""
+ python -um pip install tensorboard torch-tb-profiler
+
+Start the TensorBoard server (default at port 6006)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""
+ tensorboard --logdir ./tensorboard --port 6006
+
+Now open the following url on your browser
```
Comment (awaelchli) [inline docs/source/intermediate.rst:190]:
```suggestion
Now open the following url in your browser
```

**Review by awaelchli — APPROVED** [review]

File: docs/source/intermediate.rst [hunk 6]
```diff
@@ -97,3 +97,118 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+    1. Open Chrome and copy/paste this URL: `chrome://tracing/`.
+    2. Once tracing opens, click on `Load` at the top-right and load one of the generated traces.
+
+Limitations
+^^^^^^^^^^^^
+
+- When using the HPUProfiler, wall clock time will not be representative of the true wall clock time. This is due to forcing profiled operations to be measured synchronously, when many HPU ops happen asynchronously.
+  It is recommended to use this Profiler to find bottlenecks/breakdowns, however for end to end wall clock time use the SimpleProfiler.
+
+- HPUProfiler.summary() is not supported
```
Comment (Borda) [inline docs/source/intermediate.rst:212]:
```suggestion
- ``HPUProfiler.summary()`` is not supported
```

File: docs/source/intermediate.rst [hunk 7]
```diff
@@ -97,3 +97,118 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+    from pytorch_lightning import Trainer
+    from lightning_habana.pytorch.profiler.profiler import HPUProfiler
+
+    profiler = HPUProfiler(filename="perf-logs")
+    trainer = Trainer(profiler=profiler, accelerator="hpu")
+
+Custom Profiling
+^^^^^^^^^^^^^^^^^
+
+To [profile custom actions of interest](https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_expert.html#profile-custom-actions-of-interest), reference a profiler in the LightningModule:
```
Comment (Borda) [inline docs/source/intermediate.rst:138]:
```suggestion
To `profile custom actions of interest <https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_expert.html#profile-custom-actions-of-interest>`_, reference a profiler in the LightningModule:
```

File: docs/source/intermediate.rst [hunk 8]
```diff
@@ -97,3 +97,118 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+            print("do somthing")
+            self.profiler.stop("my_custom_action")
+            return data
+
+    # Pass profiler instance to LightningModule
+    profiler = HPUProfiler()
+    model = MyModel(profiler)
+    trainer = Trainer(accelerator="hpu", profiler=profiler)
+
+For more details on profiler, refer to [PyTorchProfiler](https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_intermediate.html)
```
Comment (Borda) [inline docs/source/intermediate.rst:170]:
```suggestion
For more details on profiler, refer to `PyTorchProfiler <https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_intermediate.html>`_
```

File: docs/source/intermediate.rst [hunk 9]
```diff
@@ -97,3 +97,118 @@ The below snippet shows how DeviceStatsMonitor can be enabled.
...
+    from pytorch_lightning import Trainer
+    from lightning_habana.pytorch.profiler.profiler import HPUProfiler
+
+    profiler = HPUProfiler(filename="perf-logs")
+    trainer = Trainer(profiler=profiler, accelerator="hpu")
+
+Custom Profiling
+^^^^^^^^^^^^^^^^^
+
+To `profile custom actions of interest <https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_expert.html#profile-custom-actions-of-interest>`_, reference a profiler in the LightningModule:
```
Comment (Borda) [inline docs/source/intermediate.rst:138]:
```suggestion
To `profile custom actions of interest <https://pytorch-lightning.readthedocs.io/en/stable/tuning/profiler_expert.html#profile-custom-actions-of-interest>`_, reference a profiler in the ``LightningModule``:
```

**Review by Borda — APPROVED** [review]

**Pull request merged**